src/tests/ftest/deployment/disk_failure.py

```diff
@@ -0,0 +1,186 @@
+        """
+        self.dmg_command.json.value = True
+        try:
+            result = self.dmg_command.storage_query_list_devices()
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+        finally:
+            self.dmg_command.json.value = False
+        data = json.loads(result.stdout_text)
+        resp = data['response']
+        if data['error'] or len(resp['host_errors']) > 0:
+            if data['error']:
+                self.fail("dmg command failed: {}".format(data['error']))
+            else:
+                self.fail("dmg command failed: {}".format(resp['host_errors']))
+        device_info = {}
+        count = 0
+        for value in list(resp['host_storage_map'].values()):
+            if value['storage']['smd_info']['devices']:
+                for target in range(self.targets):
+                    device_info[count] = {}
+                    uuid_value = value['storage']['smd_info']['devices'][target]['uuid']
+                    device_info[count]["uuid"] = uuid_value
+                    rank = value['storage']['smd_info']['devices'][target]['rank']
+                    device_info[count]["rank"] = rank
+                    tgt_ids = value['storage']['smd_info']['devices'][target]['tgt_ids']
+                    device_info[count]["tgts"] = tgt_ids
+            count = count + 1
+        return device_info
+
+    def set_nvme_faulty(self, nvme_id=None):
+        """Get a device to faulty state.
+
+        Args:
+            device_id (str): Device UUID
+        Returns:
+            resp (json format) : dmg device faulty information.
+        """
+        data = {}
+        if nvme_id is None:
+            self.fail("No device id provided")
+        self.dmg_command.json.value = True
+        try:
+            result = self.dmg_command.storage_set_faulty(uuid=nvme_id)
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+        finally:
+            self.dmg_command.json.value = False
+        data = json.loads(result.stdout_text)
+        resp = data['response']
+        return resp
+
+    def verify_disk_failure(self, num_pool):
+        """Run IOR and create disk failures while IO is happening.
+
+        Args:
+            num_pool : Total number of pools to run the testing.
+        """
+        device_info = {}
+        pool = {}
+
+        # Get the device information.
+        device_info = self.get_nvme_device_info()
+        self.log.info("Device information")
+        self.log.info("==================")
+        self.log.info(device_info)
+        for val in range(0, num_pool):
+            pool[val] = add_pool(self, connect=False)
+
+        for val in range(0, num_pool):
+            threads = []
+            self.pool = pool[val]
+            # The following thread runs while raising disk faults
+            threads.append(threading.Thread(target=self.run_ior_thread,
```

Comment:
Could these loops be combined?

Reply:
I think we can do it... Since the test is done in sequence on each pool. If the test is done in parallel, then probably not... I can combine them.